docs/DeveloperGuide.md

```diff
@@ -206,9 +207,57 @@ This section describes some noteworthy details on how certain features are imple
 #### What it is
 #### Implementation details
 
-### Match feature
+### Matching Feature
```

Comment:
Is there a reason why you changed the header? :o I think it became inconsistent with the other PRs' headers 😦